Octokit/Http/SimpleJsonSerializer.cs

```diff
@@ -190,6 +190,18 @@ public override object DeserializeObject(object value, Type type)
                     return base.DeserializeObject(value, payloadType);
                 }
 
+                if (ReflectionUtils.IsStringEnumWrapper(type))
+                {
+                    // this check is a workaround for https://github.com/octokit/octokit.net/issues/2052
+                    // as the API is returning a null value where the enum is
+                    // expecting something like a string
+                    //
+                    // this should be removed once we can confirm the GitHub API
+                    // is no longer returning a null for the parent Team's
+                    // permission value
+                    return Activator.CreateInstance(type, "null");
```

Comment:
I used `null` in my code s well as for the ParameterAttribute for the enum. `PermissionLevel` appears to use the word `none` as its value though. Do you need to change this to be `none` instead of `null`?

Reply:
I prefer `null` here to explicitly indicate no value was found, and I want this to be different to the `none` scenario that might be valid here. 

In the test itself you'll see that I can't parse out a valid value from this enum.

```C#
Assert.Equal("null", result.Parent.Permission.StringValue);

PermissionLevel value;
Assert.False(result.Parent.Permission.TryParse(out value));
```

This fixes the deserialization issue, and leverages our existing string-like enums to handle unknown values rather than mapping this `null` to `none`.

Reply:
Ahha, that makes more sense, thanks!